hazelcast/src/test/java/com/hazelcast/collection/impl/queue/QueueCapacityTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.collection.impl.queue;
+
+import com.hazelcast.collection.IQueue;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.QueueConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class  QueueCapacityTest {
+    private static final String CAPACITY_QUEUE = "TEST_QUEUE";
+    private static final int QUEUE_CAPACITY = 2;
+
+    private static HazelcastInstance instance;
+
+    @Before
+    public void setup() {
+        Config memberConfig = new Config();
+        memberConfig.addQueueConfig(new QueueConfig().setName(CAPACITY_QUEUE).setMaxSize(QUEUE_CAPACITY));
+        instance = Hazelcast.newHazelcastInstance(memberConfig);
+    }
+
+    @Test
+    public void testQueueLimitEffective() {
+        final IQueue<String> queue = instance.getQueue(CAPACITY_QUEUE);
+        final List<String> collection = Collections.nCopies(20, "Hello");
+        final boolean added = queue.addAll(collection);
+        Assert.assertFalse(added);
```

Comment:
This is actually incorrect. The contract of `java.util.Collection#addAll` says:
```
     * @throws IllegalStateException if not all the elements can be added at
     *         this time due to insertion restrictions
```

Reply:
Hi @frant-hartm 
This actually not call java.util.Collection#addAll, but call by com.hazelcast.collection.impl.queue.QueueProxyImpl#addAll
It will return false if all the elements can not be added. 

https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueProxyImpl.java

`    @Override
    public boolean addAll(@Nonnull Collection<? extends E> es) {
        checkNotNull(es, "Null collection is not allowed!");

        return addAllInternal(toDataList(es));
    }
`

Reply:
What I meant is that if you can't add all the elements due to some restriction (like exceeding the capacity) an `IllegalStateException` should be thrown, the test tests it returns false.

This is incorrect according to the contract of `Collection#addAll`, which we generally don't change, unless there is a very good reason to do so.

Reply:
Hi @frant-hartm 
I got what you mean, the previous logic returned the false. I changed that logic to throw IllegalStateException instead of return false, and push another commit.